src/core/dbus-unit.c

```diff
@@ -951,6 +965,7 @@ const sd_bus_vtable bus_unit_vtable[] = {
         SD_BUS_PROPERTY("InvocationID", "ay", bus_property_get_id128, offsetof(Unit, invocation_id), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
         SD_BUS_PROPERTY("CollectMode", "s", property_get_collect_mode, offsetof(Unit, collect_mode), SD_BUS_VTABLE_PROPERTY_CONST),
         SD_BUS_PROPERTY("Refs", "as", property_get_refs, 0, 0),
+        SD_BUS_PROPERTY("ActivationDetails", "a(ss)", property_get_activation_details, 0, SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

Comment:
Use the offset parameter to the macro (i.e. 4th arg, where you currently pass 0), so that you can use the same implementation for the property_get_activation_details() of the unit and the job. If you pass the offset to the `activation_details` field in the Unit/Job structure there, then both just have to deref a pointer to a pointer to get to the struct.

i.e. here it would be:

```c
SD_BUS_PROPERTY("ActivationDetails", "a(ss)", busproperty_get_activation_details, offsetof(Unit, activation_details), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```

and over in the job dbus code:

```c
SD_BUS_PROPERTY("ActivationDetails", "a(ss)", busproperty_get_activation_details, offsetof(Job, activation_details), SD_BUS_VTABLE_PROPERTY_EMITS_CHANGE),
```